include/engine/guidance/collapse_scenario_detection.hpp

```diff
@@ -36,7 +36,10 @@ bool isStaggeredIntersection(const RouteStepIterator step_prior_to_intersection,
 // a - > x
 bool isUTurn(const RouteStepIterator step_prior_to_intersection,
              const RouteStepIterator step_entering_intersection,
-             const RouteStepIterator step_leaving_intersection);
+             const RouteStepIterator step_leaving_intersection,
+             const std::string &step_prior_name,
+             const std::string &step_entering_name,
+             const std::string &step_leaving_name);
```

Comment:
Why do we need the strings here? Aren't they in the route steps already?

Reply:
oo good catch, did not see that this was already available.